code/modules/logic/converter.dm

```diff
@@ -0,0 +1,143 @@
+			user << "<span class='warning'>The [S] is already secured.</span>"
+			return
+		if(attached_signaler)
+			user << "<span class='warning'>There is already a device attached, remove it first.</span>"
+			return
+		user.unEquip(S)
+		S.forceMove(src)
+		S.holder = src
+		S.toggle_secure()
+		if(logic_output)		//Make sure we are set to receive if the converter is set to output logic, and send if the converter is set to accept logic input
+			S.receiving = 1
+		else
+			S.receiving = 0
+		attached_signaler = S
+		user << "<span class='notice'>You attach \the [S] to the I/O connection port and secure it.</span>"
+		return
+	if(attached_signaler && istype(O, /obj/item/weapon/screwdriver))		//Makes sure we remove the attached signaler before we can open up and deconstruct the machine
+		var/obj/item/device/assembly/signaler/S = attached_signaler
+		attached_signaler = null
+		S.forceMove(get_turf(src))
+		S.holder = null
+		S.toggle_secure()
+		user << "<span class='notice'>You unsecure and detach \the [S] from the I/O connection port.</span>"
+		return
+	..()
+
+/obj/machinery/logic_gate/convert/multitool_menu(var/mob/user, var/obj/item/device/multitool/P)
+	var/logic_state_string
+	var/menu_contents = {"
+	<dl>
+	"}
+	if(logic_output)
+		switch(output_state)
+			if(LOGIC_OFF)
+				logic_state_string = "OFF"
+			if(LOGIC_ON)
+				logic_state_string = "ON"
+			if(LOGIC_FLICKER)
+				logic_state_string = "FLICKER"
+			else
+				logic_state_string = "ERROR: UNKNOWN STATE"
+		menu_contents += {"
+		<dt><b>Output:</b> [format_tag("ID Tag","output_id_tag")]</dt>
+		<dd>Output State: [logic_state_string]</dd>
+		"}
+	else
+		switch(input1_state)
+			if(LOGIC_OFF)
+				logic_state_string = "OFF"
+			if(LOGIC_ON)
+				logic_state_string = "ON"
+			if(LOGIC_FLICKER)
+				logic_state_string = "FLICKER"
+			else
+				logic_state_string = "ERROR: UNKNOWN STATE"
+		menu_contents += {"
+		<dt><b>Input:</b> [format_tag("ID Tag","input1_id_tag")]</dt>
+		<dd>Input State: [logic_state_string]</dd>
+		"}
+	menu_contents += {"
+	<dt><b>Logic Signal Designation:</b> <a href='?src=\ref[src];toggle_logic=1'>[logic_output ? "Output" : "Input"]</a></dt>
+	"}
+	if(attached_signaler)
+		menu_contents += "<dt><a href='?src=\ref[src];adjust_signaler=1'>Adjust Signaler Settings</a></dt>"
+	else
+		menu_contents += "<dt><b>NO SIGNALER ATTACHED!</b></dt>"
+	menu_contents += {"
+	</dl>
+	"}
+	return menu_contents
+
+/obj/machinery/logic_gate/convert/multitool_topic(var/mob/user,var/list/href_list,var/obj/O)
+	..()
+	if("toggle_logic" in href_list)
+		logic_output = !logic_output
+		if(attached_signaler)		//If we have a signaler attached, make sure we update it to send/receive when we change the logic signal desgination
+			if(logic_output)
+				attached_signaler.receiving = 1
+			else
+				attached_signaler.receiving = 0
+	if("adjust_signaler" in href_list && attached_signaler)		//Make sure that we have a signaler attached to handle this one, otherwise ignore this command
```

Comment:
I think this needs to be `if(("adjust_signaler" in href_list) && attached_signaler)`